component/rbac.libsonnet

```diff
@@ -0,0 +1,64 @@
+local kap = import 'lib/kapitan.libjsonnet';
+local kube = import 'lib/kube.libjsonnet';
+local inv = kap.inventory();
+local params = inv.parameters.control_api;
+
+local defaultAnnotations = {
+  syn_component: inv.parameters._instance,
+};
+
+/*
+* Allows discovery of services, pods, and endpoints for prometheus auto-discovery
+*/
+local discoveryRole = kube.ClusterRole('syn-prometheus-auto-discovery') {
+  metadata+: {
+    annotations+: defaultAnnotations,
+  },
+  rules: [
+    {
+      apiGroups: [
+        '',
+      ],
+      resources: [
+        'pods',
+        'services',
+        'endpoints',
+      ],
+      verbs: [
+        'get',
+        'list',
+        'watch',
+      ],
+    },
+    {
+      apiGroups: [
+        'networking.k8s.io',
+      ],
+      resources: [
+        'ingresses',
```

Comment:
what do we need ingresses for here?

Reply:
I don't think we really need them right now. But prometheus recommends this: https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/rbac.md#prometheus-rbac  And for example the `prometheus-k8s` role in the `default` namspace also has it.  Probably for some auto-discovery feature we're not using right now

Reply:
fair enough, might be worth it to add a link to that page as a comment above the clusterrole